linera-core/src/updater.rs

```diff
@@ -295,6 +297,28 @@ where
                     // synchronize them now and retry.
                     self.send_chain_information_for_senders(chain_id).await?;
                 }
+                Err(NodeError::EventsNotFound(event_ids)) => {
```

Comment:
We come here with the same pattern that we have seen before.
Recoverable errors and unrecoverable errors.
Yet, in order to organize the code better, we should have a nested of error like `Result<Result<_,_>,_>`.
But we do not want that. Or maybe we could have a triple
```rust
enum OurResult {
  ok: T,
  recoverable: K,
  unrecoverable: L
}
```

I think we can gain a little by being clearer about this aspect of the code.

Reply:
Yes, but reorganizing our errors will be a huge change, much bigger than this PR.

Reply:
Ok, course. Yes, not in this PR